lib/stack_master/parameter_resolvers/stack_output.rb

```diff
@@ -34,21 +34,33 @@ def cf
         @cf ||= StackMaster.cloud_formation_driver
       end
 
-      def validate_value!(value)
-        if !value.is_a?(String) || !value.include?('/')
-          raise ArgumentError, 'Stack output values must be in the form of stack-name/output-name'
+      def parse!(value)
+        if !value.is_a?(String) || !(match = @output_regex.match(value))
+          raise ArgumentError, 'Stack output values must be in the form of [region:]stack-name/output_name'
         end
+
+        match.captures
       end
 
-      def find_stack(stack_name)
-        @stacks.fetch(stack_name) do
+      def find_stack(stack_name, region)
+        unaliased_region = @config.unalias_region(region)
+        stack_key = stack_key(stack_name, unaliased_region)
+
+        @stacks.fetch(stack_key) do
+          original_region = cf.region
+
+          cf.set_region(unaliased_region) if region && original_region != unaliased_region
```

Comment:
Instead of changing the region of `AwsDriver::CloudFormation`, how about creating, and caching, a different instance per region instead? This way, you don't need to worry about reseting the `region` value and subsequently invaliding the underlying `Aws::CloudFormation::Client` for every stack output resolved.